app/services/hyrax/lease_manager.rb

```diff
@@ -47,6 +62,51 @@ def release_lease_for!(resource:, query_service: Hyrax.query_service)
         new(resource: resource, query_service: query_service)
           .release!
       end
+
+      # Creates or updates an existing lease on a member to match the lease on the parent work
+      # @param [Array<Valkyrie::Resource>] members
+      # @param [Hyrax::Work] work
+      # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+      def create_or_update_lease_on_members(members, work)
```

Comment:
putting this method in the LeaseManger feels like a better place for it to be than in "transactions/save.rb". it also makes it easier to use when #6131 gets worked on.